File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Calling Azure Services from Web Parts/VB/AZURE_CallingWebPart/CallingWebPart/CallingWebPart.vb
Imports System
Imports System.ComponentModel
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.WebControls
Imports System.ServiceModel

<ToolboxItemAttribute(false)> _
Public Class CallingWebPart
    Inherits WebPart

    'UI Controls
    Private instructionsLabel As Label
    Private resultLabel As Label
    Private buttonToday As Button

    Protected Overrides Sub CreateChildControls()
        'Set up the instructions label
        instructionsLabel = New Label()
        instructionsLabel.Text = "Click the Get Today button to call the Azure service"
        Me.Controls.Add(instructionsLabel)
        'Set up the button
        buttonToday = New Button()
        buttonToday.Text = "Get Today"
        AddHandler buttonToday.Click, AddressOf buttonGetToday_Click
        Me.Controls.Add(buttonToday)
        'Set up the results label
        resultLabel = New Label()
        Me.Controls.Add(resultLabel)
    End Sub

    Protected Overrides Sub Render(ByVal writer As System.Web.UI.HtmlTextWriter)
        'Render the controls
        instructionsLabel.RenderControl(writer)
        writer.Write("<br />")
        buttonToday.RenderControl(writer)
        writer.Write("<br />")
        resultLabel.RenderControl(writer)
    End Sub

    Private Sub buttonGetToday_Click(ByVal sender As Object, ByVal args As EventArgs)
        'I used svcutil.exe to generate the proxy class for the service
        'in the generatedDayNamerProxy.cs file. I'm going to configure this
        'in code by using a channel factory.

        'Create the channel factory with a Uri, binding and endpoint
        'The Uri is to the hosted service in Windows Azure
        'Edit this Uri to match the location where you published the service.
        Dim serviceUri As Uri = New Uri("http://daynamervb.cloudapp.net/dayinfoservice.svc")
        Dim serviceBinding As BasicHttpBinding = New BasicHttpBinding()
        Dim dayNamerEndPoint As EndpointAddress = New EndpointAddress(serviceUri)
        Dim channelFactory As ChannelFactory(Of IDayNamer) = New ChannelFactory(Of IDayNamer)(serviceBinding, dayNamerEndPoint)
        'Create a channel
        Dim dayNamer As IDayNamer = channelFactory.CreateChannel()
        'Now we can call the TodayIs method
        Dim today As String = dayNamer.TodayIs()
        resultLabel.Text = "Today is: " + today
        'close the factory with all its channels
        channelFactory.Close()
    End Sub

End Class

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Calling Azure Services from Web Parts/C#/AZURE_CallingWebPart/CallingWebPart/CallingWebPart.cs
using System;
using System.ComponentModel;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using Microsoft.SharePoint;
using Microsoft.SharePoint.WebControls;
using System.ServiceModel;

namespace AZURE_CallingWebPart.CallingWebPart
{
    //This web part calls the DayInfoService.svc WCF service in Windows Azure
    //Make sure you package and publish the service in your Window Azure account
    //before you run this web part.
    [ToolboxItemAttribute(false)]
    public class CallingWebPart : WebPart
    {
        Label instructionsLabel;
        Button getTodayButton;
        Label resultsLabel;

        protected override void CreateChildControls()
        {
            //Set up the instructions label
            instructionsLabel = new Label();
            instructionsLabel.Text = "Click the Get Today button to call the Azure service";
            this.Controls.Add(instructionsLabel);
            //Set up the GetToday button
            getTodayButton = new Button();
            getTodayButton.Text = "Get Today";
            getTodayButton.Click += new EventHandler(getTodayButton_Click);
            this.Controls.Add(getTodayButton);
            //Set up the results label
            resultsLabel = new Label();
            this.Controls.Add(resultsLabel);
        }

        protected override void Render(HtmlTextWriter writer)
        {
            //Render the instructions label
            instructionsLabel.RenderControl(writer);
            writer.Write("<br />");
            //Render the button
            getTodayButton.RenderControl(writer);
            writer.Write("<br />");
            //Render the results label
            resultsLabel.RenderControl(writer);
        }

        private void getTodayButton_Click(object sender, EventArgs args)
        {
            //I used svcutil.exe to generate the proxy class for the service
            //in the generatedDayNamerProxy.cs file. I'm going to configure this
            //in code by using a channel factory.

            //Create the channel factory with a Uri, binding and endpoint
            //The Uri is to the hosted service in Windows Azure
            //Edit this Uri to match the location where you published the service.
            Uri serviceUri = new Uri("http://daynamercs.cloudapp.net/dayinfoservice.svc");
            BasicHttpBinding serviceBinding = new BasicHttpBinding();
            EndpointAddress dayNamerEndPoint = new EndpointAddress(serviceUri);
            ChannelFactory<IDayInfo> channelFactory = new ChannelFactory<IDayInfo>(serviceBinding, dayNamerEndPoint);
            //Create a channel
            IDayInfo dayNamer = channelFactory.CreateChannel();
            //Now we can call the TodayIs method
            string today = dayNamer.TodayIs();
            resultsLabel.Text = "Today is: " + today;
            //close the factory with all its channels
            channelFactory.Close();
        }
    }
}